src/dmd/escape.d

```diff
@@ -1054,40 +1054,39 @@ bool checkNewEscape(Scope* sc, Expression e, bool gag)
         /* Check for returning a ref variable by 'ref', but should be 'return ref'
          * Infer the addition of 'return', or set result to be the offending expression.
          */
-        if (v.storage_class & (STC.ref_ | STC.out_))
+        if (!(v.storage_class & (STC.ref_ | STC.out_)))
+            continue;
+
+        // DIP25
```

Comment:
That's not quite correct anymore. The comment DIP25 applied to the block under.
Now that I look at this, it does look a bit weird. I suppose the rationale for skipping DIP25 on non-root module is that we don't do semantic on functions body if we don't compile them (unless they're `auto`). But then we shouldn't be in this function to begin with.

Reply:
Hmm, do you think the code comment below is fine on its own? If so I'll just remove this.

Reply:
The code in escape.d is pretty sensitive, please make darn sure that a refactoring here does NOT introduce ANY behavioral changes.

Reply:
This particular is just comment placement. Previously it was a comment followed by an `if` block. This PR changes the `if` to an early continue.

Point taken though.